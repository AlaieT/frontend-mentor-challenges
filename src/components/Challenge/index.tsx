import React from "react";

import type { ChallengeProps } from "../../types";

import { ReactComponent as ArrowIcon } from "../../assets/images/arrow_outward.svg";

import styles from "../../styles/components/challenge.module.scss";

const Challenge = ({ name, link }: ChallengeProps) => (
  <a
    className={styles.challenge}
    href={link}
    target="_blank"
    rel="noopener noreferrer"
  >
    <p>{name}</p>
    <div className={styles.arrow}>
      <ArrowIcon width={24} height={24} />
    </div>
  </a>
);

export default Challenge;
