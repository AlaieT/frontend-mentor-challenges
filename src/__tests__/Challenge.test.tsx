import React from "react";
import { render, cleanup } from "@testing-library/react";

import Challenge from "../components/Challenge";

afterEach(cleanup);

describe("Header", () => {
  const props = {
    name: "test project",
    link: "https://some-link.com"
  };

  describe("correct render", () => {
    it("should match snapshot", () => {
      expect(
        render(<Challenge name={props.name} link={props.link} />).asFragment()
      ).toMatchSnapshot();
    });
  });
});
