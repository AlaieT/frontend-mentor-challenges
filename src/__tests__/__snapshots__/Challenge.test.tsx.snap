// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Challenge correct render should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="challenge"
  >
    <img
      alt="preview Some challenge"
      src="https://some-img.png"
    />
    <div
      class="name"
    >
      <p>
        Some challenge
      </p>
      <p>
        000-
      </p>
    </div>
  </div>
</DocumentFragment>
`;
